src/rust/ide/src/double_representation/definition.rs

```diff
@@ -132,6 +132,34 @@ impl DefinitionInfo {
         Self::from_line_ast(ast,kind)
     }
 
+    /// Gets the definition block lines.
+    pub fn block_lines(&self) -> FallibleResult<Vec<ast::BlockLine<Option<Ast>>>> {
+        if let Ok(block) = known::Block::try_from(self.ast.rarg.clone()) {
```

Comment:
What is this if about? Please never, ever create such complex expressions in if. Create a variable to tell what this if is checking.

Reply:
It is complex?

Well, I see on the first glance what's going on: I see try_from function, so it's likely try convert ast to block and convert it.

Maybe only self.ast.rarg.clone could be in another line.

Reply:
I'm with @farmaazon here — this feels quite simple to me.

> Maybe only self.ast.rarg.clone could be in another line.

Actually this should be replaced by `self.body()`.